server/proxy_auth.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2021 The Rode Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"context"
+	"github.com/grpc-ecosystem/go-grpc-middleware/util/metautils"
+	"google.golang.org/grpc/credentials"
+)
+
+type ProxyAuth struct {
+	requireTransportSecurity bool
+}
+
+func NewProxyAuth(requireTransportSecurity bool) credentials.PerRPCCredentials {
```

Comment:
I wonder if this should exist in the common package since this will be the second collector implementing this.